Add script organization slides (#84)
AlexsLemonade/reproducible-research: instructor_notes/06_script-content-and-stitching.md

This document provides instructions for teaching about the script contents, followed by how to use shell scripting to automatically run related scripts in order, "stitched" together.

## Learning goals

At the end of this activity, workshop attendees should be able to:

* Examine the structure of provided shell scripts
* Run R scripts with arguments from the command line, including the use of `--help`
* Render Rmarkdown documents/notebooks on the command line
* Understand how shell scripts can be used to automate multi-step analyses
* Recognize some limitations of shell scripting to automate analyses

## Activity Part 1: Exploring & running the R script & notebook

At this point we will have completed a slide presentation that describes the R script contents.
This section is partially a review/reiteration of that content, followed by running the script on the command line.

* In RStudio, have trainees open `analyses/mutation_counts/01_count-gene-mutations.R`
* Briefly walk through the components of the R script, including the following:
  * header comments
  * package imports
  * option setting & error checking
  * constants (note that in this case these are set after options)

> * During the above (notably option setting) demonstrate usage of the r script

Reviewer: ```suggestion
* During the above (notably option setting) demonstrate usage of the R script
```

  * Point out optparse section, showing section for each flag
  * Show the help feature you get for free with options:

  ```sh
  cd analyses/mutation_counts # if needed

  Rscript 01_count-gene-mutations.R -h
  ```
  * Show the command running with correct options

  ```sh

>   Rscript 01_count-gene-mutations.R --maf ../../data/processed/maf_files/LGAT.maf.tsv.gz -o LGAT_gene-mutations.tsv

Reviewer: Question about this - Are we intentionally showing them the two different syntaxes (synti? syntaxi?!?!) for providing the arguments? Would be worth pointing out either way that you can mix and match.

  ```

>   * Possibly: include a typo in the MAF file, and show the error message!

Reviewer: Always love a good intentional mistake.

> * Have the trainees open the rendered notebook `analyses/mutation_counts/01_count-gene-mutations.nb.html` in the browser

Reviewer: ```suggestion
* Have the trainees open the rendered notebook `analyses/mutation_counts/02_mutation-count-plots.nb.html` in the browser.
```

>   * Point out the lovely rendering that we get for free!

Reviewer: Worth emphasizing too that an HTML file **is simply a file.** It's just a file that opens in the browser apps but it's not always the internet. HTML files can be shared and opened without regard to internet connection!

# Instructor Notes: Rscript content, usage & simple workflows with shell scripts

This document provides instructions for teaching about the script contents, followed by how to use shell scripting to automatically run related scripts in order, "stitched" together.

## Learning goals

At the end of this activity, workshop attendees should be able to:

* Examine the structure of provided shell scripts
* Run R scripts with arguments from the command line, including the use of `--help`
* Render Rmarkdown documents/notebooks on the command line
* Understand how shell scripts can be used to automate multi-step analyses
* Recognize some limitations of shell scripting to automate analyses

## Activity Part 1: Exploring & running the R script & notebook

> At this point we will have completed a slide presentation that describes the R script contents.

Reviewer: ```suggestion
At this point we will have completed a slide presentation that describes the R script and notebook contents.
```
